sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/LookupCatalog.scala

```diff
@@ -155,4 +155,37 @@ private[sql] trait LookupCatalog extends Logging {
         None
     }
   }
+
+  /**
+   * Extract catalog and function identifier from a multi-part name with the current catalog if
+   * needed.
+   *
+   * Note that: function is only supported in v1 catalog.
+   */
+  object CatalogAndFunctionIdentifier {
```

Comment:
Have you checked my comment? https://github.com/apache/spark/pull/28840/files#r442021563  I personally think you don't need this refactoring. Could you just use `parseSessionCatalogFunctionIdentifier` in this PR?

Reply:
Thanks for remind, missed here.

Because of the dependency that `ResolveSessionCatalog` is at sql-core package. Seems I have to do the refactor.

Reply:
Yea, moving `parseSessionCatalogFunctionIdentifier` into the catalyst package looks okay, but I meant you need to refactor `parseSessionCatalogFunctionIdentifier` as a extractor?

Reply:
I will revert related change later.

Reply:
After some thought. We have to modify the `parseSessionCatalogFunctionIdentifier`

* since it used by v2, should return both `CatalogPlugin` and `FunctionIdentifier`
* during resolve `UnresolvedFunc`, it's hard to decide the sql param. Actually the sql param is not important.

And after this, we also have to update the existing code in `ResolveSessionCatalog`. Maybe it's better to do the refactor ?

Reply:
> Actually the sql param is not important.
Sorry, but not sure what it means. I made a PR for your branch. Please check my suggestion: https://github.com/ulysses-you/spark/pull/6